### DISCUSSION TITLE
Slash command permissions

### DISCUSSION THREAD

**Discussion by Thanatoros** [discord-net/Discord.Net#2148] [Q&A] [answered]
I wonder if it is possible with discord.net somehow set slash command permissions for role/user like in Discord.js
[https://discordjs.guide/interactions/slash-commands.html#slash-command-permissions](https://discordjs.guide/interactions/slash-commands.html#slash-command-permissions)

**Comment by csmir** [comment]
@Cenngo care to give some context? For as long as the base permission system is all we have, it may be good to revisit this discussion for a clear answer

**Comment by Cenngo** [comment, marked as answer]
Both the interaction service and the core library have the necessary methods to interact with the permissions api. Using the core library functions, you can set/overrider command permissions using the ApplicationCommand.ModifyCommandPermissionsAsync() method. Using the interaction service, InteractionService.ModifySlashCommandPermissionsAsync() method allows you to modify command permissions from the info class of an attribute based slash command

**Reply by ajruckman** [reply]
Is there any way to do this through the Sockets API without using the REST API?

**Reply by jdrames** [reply]
@ajruckman -- I was running into the same request as there were slash commands that I only want the guild admin to use or users with specific permissions.
with the socket client I was able to do something like below.

```
var guild = _client.GetGuild(guildId);

var cmdBuilder = new SlashCommandBuilder()
    .WithName("admincmdname")
    .WithDescription("command only available to admins")
    .WithDefaultPermissions(false);

var cmd = await guild.CreateApplicationCommandAsync(cmdBuilder.Build());
var restGuild = await _client.Rest.GetGuildAsync(guildId);
var restCmd = await restGuild.GetApplicationCommandAsync(cmd.Id);

// Add all permissions that will be applied to this cmd here
// can target specific users or roles
var permissions = new List<ApplicationCommandPermission>();
permissions.Add(new(guild.OwnerId, ApplicationCommandPermissionTarget.User, true));

await restCmd.ModifyCommandPermissions(permissions.ToArray());
```

edited to user ownerId value from socket guild.

**Reply by Cenngo** [reply]
> Is there any way to do this through the Sockets API without using the REST API?

Websocket is almost exclusively used for recieving socket events. REST API is used for everything else, so no.
